source/server/filter_chain_manager_impl.cc

```diff
@@ -93,11 +86,12 @@ void FilterChainManagerImpl::addFilterChain(
 
 void FilterChainManagerImpl::addFilterChainForDestinationPorts(
     DestinationPortsMap& destination_ports_map, uint16_t destination_port,
-    const std::vector<std::string>& destination_ips, const std::vector<std::string>& server_names,
-    const std::string& transport_protocol, const std::vector<std::string>& application_protocols,
+    const std::vector<std::string>& destination_ips,
+    const absl::Span<const std::string* const> server_names, const std::string& transport_protocol,
+    const absl::Span<const std::string* const> application_protocols,
```

Comment:
Is there a particular reason why you've chosen to use `std::string*` over something like `std::reference_wrapper`? I assume it's not possible to put a const reference in the Span?

Reply:
The only reason is that `const absl::Span<const std::string* const>` can be converted from `const ::google::protobuf::RepeatedPtrField<::std::string>` at no cost. Let me see if `span<std::reference_wrapper>` works
Below is the signature of protobuf generated function signature.

```
const ::google::protobuf::RepeatedPtrField<::std::string>&
FilterChainMatch::application_protocols() const
```

Reply:
A reference wrapper is basically a very thinly wrapped pointer, with the logic required to copy it (the pointer, not the object). It should be basically equivalent to the pointer case in terms of actual generated code. If it doesn't work right no worries, but there is some small benefit of reference_wrapper in conveying that the reference cannot be nullable and is not owned.

Reply:
Thanks! I am writing a small test cases to see if protobuf::RepeatedPtrField<string> converts to Span<reference_wrapper<string>>. Will post the code here. As long as it works I agree reference_wrapper is the preference